File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/RequestCancellationDelegate.java
package io.miragon.blueprint.adapter.inbound.operaton;

import io.miragon.blueprint.adapter.process.CancelBikeOrderProcessApi.Variables;
import io.miragon.blueprint.application.port.inbound.RequestOrderCancellationUseCase;
import io.miragon.blueprint.domain.bike.OrderId;
import org.operaton.bpm.engine.delegate.DelegateExecution;
import org.springframework.stereotype.Component;

@Component
public class RequestCancellationDelegate extends BaseDelegate {

    private final RequestOrderCancellationUseCase useCase;

    public RequestCancellationDelegate(RequestOrderCancellationUseCase useCase) {
        this.useCase = useCase;
    }

    @Override
    protected void executeTask(DelegateExecution execution) {
        // `orderId` is handed to the cancelBikeOrder sub-process by the calling activity.
        OrderId orderId = new OrderId(
                (String) execution.getVariable(Variables.StartEventCancellationRequired.ORDER_ID.getValue()));
        boolean cancellationPossible = useCase.requestCancellation(orderId);
        execution.setVariable(Variables.ServiceTaskRequestCancellation.CANCELLATION_POSSIBLE.getValue(),
                cancellationPossible);
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/RequestCancellationDelegate.kt
package io.miragon.blueprint.adapter.inbound.operaton

import io.miragon.blueprint.adapter.process.CancelBikeOrderProcessApi.Variables
import io.miragon.blueprint.application.port.inbound.RequestOrderCancellationUseCase
import io.miragon.blueprint.domain.bike.OrderId
import org.operaton.bpm.engine.delegate.DelegateExecution
import org.springframework.stereotype.Component

@Component
class RequestCancellationDelegate(
    private val useCase: RequestOrderCancellationUseCase,
) : BaseDelegate() {

    override fun executeTask(execution: DelegateExecution) {
        // `orderId` is handed to the cancelBikeOrder sub-process by the calling activity.
        val orderId = OrderId(execution.getVariable(Variables.StartEventCancellationRequired.ORDER_ID.value) as String)
        val cancellationPossible = useCase.requestCancellation(orderId)
        execution.setVariable(Variables.ServiceTaskRequestCancellation.CANCELLATION_POSSIBLE.value, cancellationPossible)
    }
}
